src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewServiceTest.kt

```diff
@@ -666,48 +667,62 @@ class PrisonerIepLevelReviewServiceTest {
+
+      val newReview = PrisonerIepLevel(
+        id = 1L,
+        prisonerNumber = "A8765SS",
+        bookingId = 1234567L,
+        prisonId = "LEI",
+        locationId = "LEI-1-1-001",
+        reviewedBy = "TEST_STAFF1",
+        iepCode = "BAS",
+        current = true,
+        reviewTime = LocalDateTime.now().minusDays(2)
+      )
+      val oldReview1 = PrisonerIepLevel(
+        id = 3L,
+        prisonerNumber = "A1244AB",
+        bookingId = 555555L,
+        prisonId = "LEI",
+        locationId = "LEI-1-1-001",
+        reviewedBy = "TEST_STAFF1",
+        iepCode = "STD",
+        current = true,
+        reviewTime = LocalDateTime.now().minusDays(100)
+      )
+
+      val oldReview2 = PrisonerIepLevel(
+        id = 2L,
+        prisonerNumber = "A1244AB",
+        bookingId = 555555L,
+        prisonId = "LEI",
+        locationId = "LEI-1-1-001",
+        reviewedBy = "TEST_STAFF1",
+        iepCode = "BAS",
+        current = false,
+        reviewTime = LocalDateTime.now().minusDays(200),
+      )
+
       whenever(prisonerIepLevelRepository.findAllByPrisonerNumberOrderByReviewTimeDesc("A8765SS"))
         .thenReturn(
           flowOf(
-            PrisonerIepLevel(
-              prisonerNumber = "A8765SS",
-              bookingId = 1234567L,
-              prisonId = "LEI",
-              locationId = "LEI-1-1-001",
-              reviewedBy = "TEST_STAFF1",
-              iepCode = "BAS",
-              current = true,
-              reviewTime = LocalDateTime.now().minusDays(2)
-            )
+            newReview
           )
         )
+
       whenever(prisonerIepLevelRepository.findAllByPrisonerNumberOrderByReviewTimeDesc("A1244AB"))
         .thenReturn(
           flowOf(
-            PrisonerIepLevel(
-              prisonerNumber = "A1244AB",
-              bookingId = 555555L,
-              prisonId = "LEI",
-              locationId = "LEI-1-1-001",
-              reviewedBy = "TEST_STAFF1",
-              iepCode = "BAS",
-              current = false,
-              reviewTime = LocalDateTime.now().minusDays(200),
-            ),
-            PrisonerIepLevel(
-              prisonerNumber = "A1244AB",
-              bookingId = 555555L,
-              prisonId = "LEI",
-              locationId = "LEI-1-1-001",
-              reviewedBy = "TEST_STAFF1",
-              iepCode = "STD",
-              current = true,
-              reviewTime = LocalDateTime.now().minusDays(100)
-            )
+            oldReview2,
+            oldReview1
           )
         )
       prisonerIepLevelReviewService.mergedPrisonerDetails(prisonMergeEvent)
 
+      verify(prisonerIepLevelRepository).save(eq(newReview.copy(prisonerNumber = "A1244AB")))
+      verify(prisonerIepLevelRepository).save(eq(oldReview2.copy(bookingId = 1234567L, id = 0L, current = false)))
+      verify(prisonerIepLevelRepository).save(eq(oldReview1.copy(bookingId = 1234567L, id = 0L, current = false)))
+
```

Comment:
So:
1. the new reviews with the new/incorrect prisoner number are updated to have the existing/correct prisoner number
2. the old reviews with the old booking id and existing/correct prisoner number are copied (but not updated because of the `id = 0L`, right?) into reviews with existing/correct prisoner number but also new booking Id so that history for merged prisoner includes old reviews

Is that correct? Makes more sense now :)

Reply:
yes